### PR TITLE
DOC: Clarify DST rounding behavior in Timestamp/DatetimeIndex

### PR DESCRIPTION
**Pull request by mroeschke**
xref https://github.com/pandas-dev/pandas/issues/44287
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them

### CODE REVIEW

File: pandas/core/arrays/datetimelike.py [hunk 1]
```diff
@@ -1639,6 +1639,13 @@ def strftime(self, date_format: str) -> npt.NDArray[np.object_]:
     ------
     ValueError if the `freq` cannot be converted.
 
+    Notes
+    -----
+    If the timestamps have a timezone, {op}ing will take place relative to the
+    local ("wall") time and re-localized to the same timezone. When {op}ing
+    near daylight savings time, use ``nonexistent`` and ``ambiguous`` to
+    control the re-localization behavior.
```
Comment (jbrockmendel) [inline pandas/core/arrays/datetimelike.py:1647]: does this merit an example?

**Comment by jreback** [comment]
just wondering out loud if we could/should actually do this in a tz-aware way for `.round`, `.ceil`, `.floor`; e.g. are these fundamentally different ops that say subtraction that we could *always* do this w/o going ambiguous

cc @mroeschke

**Comment by mroeschke** [comment]
@jreback found discussion https://github.com/pandas-dev/pandas/pull/22560#issuecomment-419334705 and  https://github.com/pandas-dev/pandas/pull/22560#issuecomment-419684301 that rounding in a tz-aware might not actually be possible and falling back to rounding based on actual, UTC time may give unexpected results.

**Comment by jreback** [comment]
my question is can we leave in thr tz and do a round operation w/o introducing ambiguity?

**Comment by mroeschke** [comment]
Hmm since the rounding operations happen on epoch timestamps, rounding could happen on the UTC epoch timestamps and relocalized without ambiguity i.e. `.tz_convert("UTC").round(...).tz_convert(tz)`. But this flies counter to https://github.com/pandas-dev/pandas/pull/22560#issuecomment-419684301 and the notion of "round relative to the wall time"

**Comment by jreback** [comment]
@mroeschke rereading i am wondering if it's always possible in freq < day (eg hour / min) to do the round ops in local time (and not convert-round-localize) which can introduce ambiguity

**Comment by mroeschke** [comment]
Yeah for small rounding frequencies this can be achieved. Looking at the UTC offsets I think the limit is sub 15 minutes (I see a +05:45 UTC offset for example)

**Comment by jreback** [comment]
thanks @mroeschke

**Pull request merged**